NickvisionTubeConverter.Shared/Models/MediaUrlInfo.cs

```diff
@@ -0,0 +1,122 @@
+    }
+
+    /// <summary>
+    /// Gets a MediaUrlInfo from a url string
+    /// </summary>
+    /// <param name="url">The media url string</param>
+    /// <returns>A MediaUrlInfo object. Null if url invalid</returns>
+    public static async Task<MediaUrlInfo?> GetAsync(string url)
+    {
+        var pathToOutput = $"{Configuration.TempDir}{Path.DirectorySeparatorChar}output.log";
+        dynamic outFile = PythonHelpers.SetConsoleOutputFilePath(pathToOutput);
+        return await Task.Run(() =>
+        {
+            try
+            {
+                var mediaUrlInfo = new MediaUrlInfo(url);
+                using (Py.GIL())
+                {
+                    dynamic ytdlp = Py.Import("yt_dlp");
+                    var ytOpt = new Dictionary<string, dynamic>() {
+                        { "quiet", true },
+                        { "merge_output_format", "/" },
+                        { "windowsfilenames", RuntimeInformation.IsOSPlatform(OSPlatform.Windows) },
+                        { "ignoreerrors", true }
+                    };
+                    PyDict? mediaInfo = ytdlp.YoutubeDL(ytOpt).extract_info(url, download: false);
+                    if (mediaInfo == null)
+                    {
+                        outFile.close();
+                        return null;
+                    }
+                    if (mediaInfo.HasKey("entries"))
+                    {
+                        foreach (var e in mediaInfo["entries"].As<PyList>())
+                        {
+                            if (e.IsNone())
+                            {
+                                continue;
+                            }
+                            mediaUrlInfo.ParseFromPyDict(e.As<PyDict>(), true);
+                        }
+                    }
+                    else
+                    {
+                        mediaUrlInfo.ParseFromPyDict(mediaInfo);
+                    }
+                    outFile.close();
+                }
+                return mediaUrlInfo;
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e);
+                using (Py.GIL())
+                {
+                    outFile.close();
+                }
+                return null;
+            }
+        });
+    }
+
+    private void ParseFromPyDict(PyDict mediaInfo, bool isPartOfPlaylist = false)
+    {
+        var title = mediaInfo.HasKey("title") ? (mediaInfo["title"].As<string?>() ?? "Media") : "Media";
+        foreach (var c in Path.GetInvalidFileNameChars())
+        {
+            title = title.Replace(c, '_');
+        }
+        foreach (var f in mediaInfo["formats"].As<PyList>())
+        {
+            var format = f.As<PyDict>();
+            if (format.HasKey("vbr"))
+            {
+                var resolution = new VideoResolution(format["width"].As<int>(), format["height"].As<int>());
+                if (!VideoResolutions.Exists(r => r == resolution))
```

Comment:
```csharp
if(!VideoResolutions.Contains(resolution)) {...}
```
should work...

Reply:
I tried it first and it didn't work for some reason, so I changed it

Reply:
Ah, you have to implement IEquatable<VideoResolution> as well. Look at Transaction model in Denaro. 

After implementation, .Contains should work.